### PR TITLE
Update to JupyterLab 4 and JupyterLite 0.2.0

### PR DESCRIPTION
**Pull request by jtpio**
Technically the xeus python kernel works without modifications with JupyterLite `0.2.0a0`: https://github.com/jupyterlite/jupyterlite/releases/tag/v0.2.0a0

But to keep dependencies correctly defined and avoid potential breakages we should consider updating to the latest packages.

- [x] Use `jlpm` (Yarn 3) to install dependencies
- [x] Update to JupyterLab 4 packages
- [x] Update to `jupyterlite-core>=0.2.0a0`

### CODE REVIEW

**Comment by jtpio** [comment]
Before merging we might want to decide first how to handle versioning (and setup branches) so we can still make releases of the kernel for JupyterLite 0.1.x until 0.2.0 final is out.

File: CHANGELOG.md
Comment (jtpio) [inline CHANGELOG.md:5]: @martinRenou looks like the force push made the changelog entry appear in the diff while it shouldn't?

Comment (martinRenou) [reply CHANGELOG.md:5]: You may have rebased in the middle of the release that got broken. You would need to rebase again and remove this commit.

Comment (jtpio) [reply CHANGELOG.md:5]: What was broken with the release?

Comment (jtpio) [reply CHANGELOG.md:5]: Ah it looks like it was a force push to the `main` branch? (not this PR)

Comment (martinRenou) [reply CHANGELOG.md:5]: I can fix that for you if you want. I don't think you can get away from this situation with the rebase in the Github UI, you need to rebase interactively from the git CLI and remove the two commits (release and merge)

Comment (jtpio) [reply CHANGELOG.md]: ok I just rebased and the diff looks fine now.

**Comment by martinRenou** [comment]
> Before merging we might want to decide first how to handle versioning (and setup branches) so we can still make releases of the kernel for JupyterLite 0.1.x until 0.2.0 final is out.

How about we plan a jupyterlite-xeus-python 1.x release with JupyterLite 0.2.0 support? So we get the full semver range afterwards.

Also, what would be the ETA for JupyterLite 0.2.0? Should we keep this PR open for now, waiting for this final release?

**Comment by jtpio** [comment]
> How about we plan a jupyterlite-xeus-python 1.x release with JupyterLite 0.2.0 support? So we get the full semver range afterwards.

That would indeed be useful.

> Also, what would be the ETA for JupyterLite 0.2.0? Should we keep this PR open for now, waiting for this final release?

There are some remaining issues listed in https://github.com/jupyterlite/jupyterlite/issues/1141. Then some other items like decoupling storage, but these could also go in a 0.3.0, so 0.2.0 is mostly focused on the lab 4 / nb 7 update.

Also we'll probably need a release of `jupyterlite-xeus-python` that supports JupyterLite 0.2.0 to be able to test it, now that https://github.com/jupyterlite/xeus-python-kernel/pull/160 is in.

**Comment by martinRenou** [comment]
Sure! So let's merge this one and make an 0.x branch + start making 1.x alpha releases.

**Comment by jtpio** [comment]
Also Meeseeks is enabled on the `jupyterlite` org, which should make it easier to do backports on branches.

**Review by martinRenou — APPROVED** [review]
Thank you!

**Comment by martinRenou** [comment]
meeseeksdev backport to 0.9.x

**Comment by martinRenou** [comment]
meeseeksdev backport to 0.x

**Pull request merged**
